### PR TITLE
UDP Relay/Headend

### PR DESCRIPTION
**Pull request by eol-account**

### CODE REVIEW

File: dome-headend/dome-udp-server.ipv6.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			self.controllers[controller].start_index = config['controllers'][controller]['start_index']
+			self.controllers[controller].ip = config['controllers'][controller]['ip']
+
+		for led in range(numLeds):
+			newLed = Led()
+			newLed.x = config['led_list'][led]['x']
+			newLed.y = config['led_list'][led]['y']
+			newLed.z = config['led_list'][led]['z']
+			self.led_list[led] = newLed # TODO: Replace with ipv6
+		print self.led_list
```
Comment (samdbmg) [inline dome-headend/dome-udp-server.ipv6.py:43]: Doesn't DomeConfig.py already do this?

File: dome-headend/dome-udp-server.ipv6.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+	def process_command(self, command, target):
+		log.debug("%r" % (command,))
+		log.debug("%r" % (target,))
```
Comment (samdbmg) [inline dome-headend/dome-udp-server.ipv6.py:47]: I'm confused - I assumed this file was just the IPv6 flaviour of dome-udp-server.py - but it doesn't appear to do the same thing?

File: dome-headend/dome-udp-server.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		if len(command) == (self.numLeds * 3):
+			log.debug("%r" % ("Valid",))
+			sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+			for currentController in range(self.numControllers):
+				start = int(self.controllers[currentController].start_index)
+				end = int(self.controllers[currentController].start_index + self.controllers[currentController].num_leds)
+				sock.sendto(command[start:end], (self.controllers[currentController].ip, self.controllerPort))
+
+		self.domejsSender.send(command, socket.AF_INET)
+		self.addressableLEDSender.send(bytearray("$") + command, socket.AF_INET6)
```
Comment (samdbmg) [inline dome-headend/dome-udp-server.py:46]: I assume this updates the "IPv6 address for eery LED" state? But I can't find what this is supposed to send to, nothing listens of port 4242. File missing or am I being thick?

**Review by samdbmg — APPROVED** [review]
Noice, seems to work.
No idea if you actually wanted a review or you just don't have write access mind...

File: dome-headend/dome-udp-server.ipv6.py (hunk 1 above)
Comment (eol-account) [reply dome-headend/dome-udp-server.ipv6.py:43]: Yeah, all of the ipv6 stuff is WIP. Probably going to be completely rewritten

**Pull request merged**